### ISSUE TITLE
Result of XML concatenation where one operand is the empty XML value

### ISSUE THREAD

**Issue by MaryamZi** [ballerina-platform/ballerina-spec#944]
**Description:**
1. For xml concatenation, the [spec](https://ballerina.io/ballerina-spec/spec.html#additive-expr) says

> if both operands are xml, then the result is a new xml value that is the concatenation of the operands; this does not perform a copy on the constituents of the operand values

but should a new xml value (sequence) be created even when one of the operands is the empty XML value? 

 For string and xml, the spec does say

 > if one operand is an empty string and one is xml, then the result is the xml value operand

2. when one of the operands is of static type `xml<never>` and the other is of static type `T` (where `T` is a subtype of xml), can't the static type of the additive expression be `T`? (could be related to https://github.com/ballerina-platform/ballerina-spec/issues/937).

For 

```ballerina
xml:Element x = xml `<elem/>`;
xml<never> y = xml ``;
```

the implementation currently allows
```ballerina
xml<xml:Element> b = x + y; // allowed
```

but not
```ballerina
xml:Element a = x + y; // error
```

**Comment by jclark** [comment]
Point 1.
 
> should a new xml value (sequence) be created even when one of the operands is the empty XML value?

The question doesn't make sense, because the only xml values that have storage identity are non-text singletons. 

Point 2.

The right  (most precise) type for `xml:Element + xml<never>` is `xml:Element`. I don't see any reason why the compiler shouldn't use that. Similarly `string:Char` + `""` should be `string:Char`.

**Comment by MaryamZi** [comment]
> Point 1.
> 
> > should a new xml value (sequence) be created even when one of the operands is the empty XML value?
> 
> The question doesn't make sense, because the only xml values that have storage identity are non-text singletons.
> 

I agree that it is the same from the user's PoV, but was wondering if we can directly say if one operand is the empty XML value, the result is the other XML operand.

jBallerina's exact equality checks with XML seem to have some issues at the moment. But once that is fixed, I believe there wouldn't be any issue/difference to the user.

```ballerina
import ballerina/io;

public function main() {
    xml a = xml ``;
    xml b = xml `<elem/>`;

    xml c = a + b;

    io:println(c === b); // false at the moment
    io:println(c.get(0) === b); // true
}
```

> Point 2.
> 
> The right (most precise) type for `xml:Element + xml<never>` is `xml:Element`. I don't see any reason why the compiler shouldn't use that. Similarly `string:Char` + `""` should be `string:Char`.

Yes, the implementation currently seems to consider the static type as `xml<xml:Element>`. 

Also, for `xml<never> + ""`, should we say the static type is `xml<never>`?
